### PR TITLE
Designed to work with my fork of terraform_validate

### PR DESCRIPTION
**Pull request by sbalbach**
The changes made to terrascan (and to a larger extent terraform_validate) were to solve several issues:
1. Don't stop on the first failure
2. Show the module and file that had the failure
3. Return 0 if no failures or errors, 4 otherwise (for implementation into a CI/CD pipeline)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -27,6 +28,49 @@ A collection of security and best practice tests for static code analysis of ter
 * Documentation: https://terrascan.readthedocs.io.
 * Free software: GNU General Public License v3
 
+--------------------
+Updates in this fork
+--------------------
+- **Requires my fork of terraform_validate and pyhcl until pull request accepted**
```
Comment (cesar-rodriguez) [inline README.rst]: Have you published your forks to pypi?

**Comment by cesar-rodriguez** [comment]
Hi @sbalbach. I'm looking to merge your changes. Do you have any plans on publishing your forks of pyhcl and terraform-validate to pypi?

**Comment by huhqwerty** [comment]
I made merge requests to pyhcl and terraform-validate.  The pyhcl project won't merge my changes because they don't work right in some cases that don't matter to terrascan.  There has been no response from the terraform-validate project.  I have updates in our company repo that I can push to the public fork.  I'll let you know when that's done.

**Comment by sbalbach** [comment]
Hi @cesar-rodriguez, I've got everything committed now.  Note that I changed the name of all three in our corporate repo to have a -sf suffix to keep them separate from the originals.  I'm not sure how you can use them but feel free.  Also, note that terrascan will return completely different results with this version.

**Comment by cesar-rodriguez** [comment]
Ok. Thank you!

**Comment by sbalbach** [comment]
Note that pyhcl just merged my changes to their master branch.  However, to use that branch would require some changes in terrascan because the type of data returned by phycl will change in certain cases.  There were issues with how pyhcl was handling quotes in comments that caused me to change how pyhcl works for terrascan's needs.

**Comment by cesar-rodriguez** [comment]
So it’s best to stick with your fork?

**Comment by huhqwerty** [comment]
At this point, yes

**Comment by sbalbach** [comment]
hmmm....I seem to have two identities...but yes, stick with my fork for now at least.

**Comment by sbalbach** [comment]
Actually, it's terraform_validate that would need to change to use the new master pyhcl.  They still haven't responded to my merge request.

**Comment by cesar-rodriguez** [comment]
Thanks again for this. I've merged it into a branch named StateFarmIns/terrascan while I work on cleaning things up, merging to master, and cutting a release for it.

**Pull request merged**